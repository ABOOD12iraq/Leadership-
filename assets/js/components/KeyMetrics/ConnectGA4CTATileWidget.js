/**
 * ConnectGA4CTATileWidget component.
 *
 * Site Kit by Google, Copyright 2023 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * External dependencies
 */
import PropTypes from 'prop-types';

/**
 * Internal dependencies
 */
import Data from 'googlesitekit-data';
import ConnectGA4CTATile from './ConnectGA4CTATile';
import { CORE_MODULES } from '../../googlesitekit/modules/datastore/constants';
const { useSelect } = Data;

export default function ConnectGA4CTATileWidget( { Widget, WidgetNull } ) {
	const isGA4ModuleConnected = useSelect( ( select ) =>
		select( CORE_MODULES ).isModuleConnected( 'analytics-4' )
	);

	if ( isGA4ModuleConnected !== false ) {
		return <WidgetNull />;
	}

	return (
		<Widget noPadding>
			<ConnectGA4CTATile />
		</Widget>
	);
}

ConnectGA4CTATileWidget.propTypes = {
	Widget: PropTypes.elementType.isRequired,
	WidgetNull: PropTypes.elementType.isRequired,
};
