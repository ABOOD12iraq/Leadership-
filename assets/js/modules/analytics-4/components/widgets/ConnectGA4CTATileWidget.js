/**
 * ConnectGA4CTATileWidget component.
 *
 * Site Kit by Google, Copyright 2023 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * External dependencies
 */
import PropTypes from 'prop-types';

/**
 * Internal dependencies
 */
import Data from 'googlesitekit-data';
import ConnectGA4CTATile from '../../../../components/KeyMetrics/ConnectGA4CTATile';
import { CORE_MODULES } from '../../../../googlesitekit/modules/datastore/constants';
import { CORE_USER } from '../../../../googlesitekit/datastore/user/constants';

const { useSelect } = Data;

export default function ConnectGA4CTATileWidget( { Widget, WidgetNull } ) {
	const isGA4ModuleConnected = useSelect( ( select ) =>
		select( CORE_MODULES ).isModuleConnected( 'analytics-4' )
	);

	const keyMetrics = useSelect( ( select ) =>
		select( CORE_USER ).getKeyMetrics()
	);

	let hideWidget = true;

	if ( keyMetrics?.length > 0 ) {
		const kmAnalyticsWidgetCount = keyMetrics.filter( ( keyMetric ) =>
			keyMetric.startsWith( 'kmAnalytics' )
		).length;
		if ( kmAnalyticsWidgetCount > 0 && kmAnalyticsWidgetCount < 3 ) {
			hideWidget = false;
		}
	}

	if ( hideWidget || isGA4ModuleConnected !== false ) {
		return <WidgetNull />;
	}

	return (
		<Widget noPadding>
			<ConnectGA4CTATile />
		</Widget>
	);
}

ConnectGA4CTATileWidget.propTypes = {
	Widget: PropTypes.elementType.isRequired,
	WidgetNull: PropTypes.elementType.isRequired,
};
