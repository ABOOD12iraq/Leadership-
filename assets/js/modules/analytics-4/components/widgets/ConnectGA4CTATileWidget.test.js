/**
 * ConnectGA4CTATileWidget component tests.
 *
 * Site Kit by Google, Copyright 2023 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * Internal dependencies
 */
import {
	provideModules,
	provideUserCapabilities,
	render,
} from '../../../../../../tests/js/test-utils';
import ConnectGA4CTATileWidget from './ConnectGA4CTATileWidget';

describe( 'ConnectGA4CTATileWidget', () => {
	it( 'should render the Connect GA4 CTA tile', () => {
		const { container, getByText } = render(
			<ConnectGA4CTATileWidget widgetSlug="some-widget" />,
			{
				setupRegistry: ( registry ) => {
					provideUserCapabilities( registry );
					provideModules( registry );
				},
			}
		);

		expect( container ).toMatchSnapshot();

		expect( getByText( 'Connect Analytics' ) ).toBeInTheDocument();
	} );
} );
