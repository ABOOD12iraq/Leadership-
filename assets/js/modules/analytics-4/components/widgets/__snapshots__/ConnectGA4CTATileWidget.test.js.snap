// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectGA4CTATileWidget should render the Connect GA4 CTA tile 1`] = `
<div>
  <div
    class="googlesitekit-widget googlesitekit-widget--keyMetricsConnectGA4CTATile"
  >
    <div
      class="googlesitekit-widget__body"
    >
      <div
        class="googlesitekit-widget--connectModuleCTATile"
      >
        <div
          class="googlesitekit-km-connect-module-cta-tile"
        >
          <div
            class="googlesitekit-km-connect-module-cta-tile__content"
          >
            <p
              class="googlesitekit-km-connect-module-cta-tile__text"
            >
              Analytics is disconnected, some of your metrics canâ€™t be displayed
            </p>
            <a
              class="googlesitekit-cta-link"
              href="#"
            >
              Connect Analytics
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
